### PR TITLE
Implement KVGraph package to allow retrieving a defined list of vm-related objects

### PR DESCRIPTION
**Pull request by alromeros**
**What this PR does / why we need it**:

This PR implements a new `kvgraph` package so we allow an exportable way to retrieve all the VM-related resources.

In this way, third-party software can use our package to retrieve the list of resources related to a specific VM:

```go

// Backup

// If we want to easily get the list of backup objects for any unstructured item
backupResources := vmgraph.NewObjectBackupGraph(item)

// Alternatively, we can pass the converted object to their specific functions
backupResources := kvgraph.NewVirtualMachineBackupGraph(vm)
backupResources := kvgraph.NewVirtualMachineInstanceBackupGraph(vmi)
backupResources := kvgraph.NewDataVolumeBackupGraph(dv)

// Restore

// If we want to easily get the list of restore objects for any unstructured item
restoreResources := kvgraph.NewObjectRestoreGraph(item)

// Alternatively, we can pass the converted object to their specific functions
vestoreResources := kvgraph.NewVirtualMachineRestoreGraph(vm)
restoreResources := kvgraph.NewVirtualMachineInstanceRestoreGraph(vmi)
```

**Which issue(s) this PR fixes** *(optional, in fixes #<issue number>(, fixes #<issue_number>, ...) format, will close the issue(s) when PR gets merged)*:
Fixes # https://issues.redhat.com/browse/CNV-41501

**Release note**:

```release-note
Implement a kvgraph package to allow retrieving a defined list of vm-related objects
```

### CODE REVIEW

File: pkg/util/graph/util.go [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+	resources = addAccessCredentials(spec.AccessCredentials, namespace, resources)
+	return resources
+}
+
+func addVolumeGraph(volumes []v1.Volume, namespace string, resources []velero.ResourceIdentifier) []velero.ResourceIdentifier {
+	for _, volume := range volumes {
+		switch {
+		case volume.DataVolume != nil:
+			resources = addVeleroResource(volume.DataVolume.Name, namespace, "datavolumes", resources)
+			resources = addVeleroResource(volume.DataVolume.Name, namespace, "persistentvolumeclaims", resources)
```
Comment (alromeros) [inline pkg/util/graph/util.go:68]: @mhenriks do we really need to explicitly backup the pvc if we already handle it in `dv_backup_item_action.go`?

File: pkg/util/graph/backup_graph.go [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+
+// NewVirtualMachineInstanceBackupGraph returns the backup object graph for a specific VMI
+func NewVirtualMachineInstanceBackupGraph(vmi *v1.VirtualMachineInstance) ([]velero.ResourceIdentifier, error) {
+	var resources []velero.ResourceIdentifier
+	var err error
+	// Returning full backup even if there was an error retrieving the launcher pod.
+	// The caller can decide wether to use the backup without launcher pod or handle the error.
+	resources, err = addLauncherPod(vmi.GetName(), vmi.GetNamespace(), resources)
+	return addCommonVMIObjectGraph(vmi.Spec, vmi.GetNamespace(), true, resources), err
+}
```
Comment (mhenriks) [inline pkg/util/graph/backup_graph.go:57]: Wonder if DataVolume should be an entrypoint as well?  Basically anything that has a a `*_backup_item_action.go`?

File: pkg/util/graph/util.go (hunk 1 above)
Comment (alromeros) [reply pkg/util/graph/util.go]: Figured out that precisely we want to be explicit as this package may be used by third-party integrations that don't use our plugin's logic.

File: pkg/util/graph/backup_graph.go (hunk 2 above)
Comment (alromeros) [reply pkg/util/graph/backup_graph.go:57]: Would that only include the pvc? maybe the populator source if it has one too? I can add it but seems a bit unnecessary, but I don't mind having it

Comment (mhenriks) [reply pkg/util/graph/backup_graph.go:57]: > Would that only include the pvc?

I think so?  Basically whatever the "extra" items we add to the backup I don't think we include populator sources.  That seems like base functionality that could/should be added to velero core

Basically I think there should be a graph function for each of our types that we specially handle

Comment (alromeros) [reply pkg/util/graph/backup_graph.go:58]: Added commit with datavolume backup graph

File: pkg/util/graph/backup_graph.go [hunk 3]
```diff
@@ -0,0 +1,99 @@
...
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * Copyright The KubeVirt Velero Plugin Authors.
+ *
+ */
+
+package vmgraph
```
Comment (alromeros) [inline pkg/util/graph/backup_graph.go:20]: @mhenriks wondering if `vmgraph` is good name for the package since it also supports other objects such as DVs.

Comment (mhenriks) [reply pkg/util/graph/backup_graph.go:20]: Yeah `objgraph` or something more generic is probably better

Curious why you chose a different package name (vmgraph) and directory (graph)?

Comment (alromeros) [reply pkg/util/graph/backup_graph.go:20]: I though `vmgraph` was better for an exportable package as sounded less generic while `graph` worked in the context of our directories, but yeah probably not the best naming decision. I'll change the package name.

Comment (alromeros) [reply pkg/util/graph/backup_graph.go:20]: What about `kvgraph`?

Comment (alromeros) [reply pkg/util/graph/backup_graph.go]: Renamed to `kvgraph`

**Pull request merged**